src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -232,141 +261,142 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
     return true;
 }
 
-bool emberAfGroupsClusterGetGroupMembershipCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
-                                                    const Commands::GetGroupMembership::DecodableType & commandData)
+struct GetGroupMembershipResponse
 {
-    auto & groupCount = commandData.groupCount;
-    auto & groupList  = commandData.groupList;
+public:
+    // Use GetCommandId instead of commandId directly to avoid naming conflict with CommandIdentification in ExecutionOfACommand
+    static constexpr CommandId GetCommandId() { return Commands::GetGroupMembershipResponse::Id; }
+    static constexpr ClusterId GetClusterId() { return Groups::Id; }
 
-    EmberStatus status = EMBER_ZCL_STATUS_FAILURE;
-    uint8_t i, j;
-    uint8_t count = 0;
-    uint8_t list[EMBER_BINDING_TABLE_SIZE << 1];
-    uint8_t listLen = 0;
-    CHIP_ERROR err  = CHIP_NO_ERROR;
+    GetGroupMembershipResponse(const Commands::GetGroupMembership::DecodableType & data,
+                               GroupDataProvider::GroupMappingIterator * iter) :
+        mCommandData(data),
+        mIterator(iter)
+    {}
 
-    emberAfGroupsClusterPrint("RX: GetGroupMembership [");
+    const Commands::GetGroupMembership::DecodableType & mCommandData;
+    GroupDataProvider::GroupMappingIterator * mIterator = nullptr;
 
-    // For all networks, Get Group Membership commands may be sent either
-    // unicast or broadcast (removing the ZLL-specific limitation to unicast).
+    uint8_t mCapacity = 0xff;
 
-    // When Group Count is zero, respond with a list of all active groups.
-    // Otherwise, respond with a list of matches.
-    // TODO: https://github.com/project-chip/connectedhomeip/issues/10335
-    if (groupCount == 0)
-    {
-        for (i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
-        {
-            EmberBindingTableEntry entry;
-            status = emberGetBinding(i, &entry);
-            if ((status == EMBER_SUCCESS) && (entry.type == EMBER_MULTICAST_BINDING) && (entry.local == emberAfCurrentEndpoint()))
-            {
-                list[listLen]     = EMBER_LOW_BYTE(entry.groupId);
-                list[listLen + 1] = EMBER_HIGH_BYTE(entry.groupId);
-                listLen           = static_cast<uint8_t>(listLen + 2);
-                count++;
-            }
-        }
-    }
-    else
+    CHIP_ERROR Encode(TLV::TLVWriter & writer, TLV::Tag tag) const
     {
-        auto iter = groupList.begin();
-        while (iter.Next())
+        TLV::TLVType outer;
+        ReturnErrorOnFailure(writer.StartContainer(tag, TLV::kTLVType_Structure, outer));
+        // A capacity of 0xFF means that it is unknown if any further groups may be
+        // added.  Each group requires a binding and, because the binding table is
+        // used for other purposes besides groups, we can't be sure what the
+        // capacity will be in the future.
```

Comment:
Again, this comment needs to be at the place where the 0xff is.  Not at the place where we do the encode, but at the place where we decide that we will be encoding the 0xff.  It makes no sense here to be talking about "0xFF", when there is no "0xFF" in sight.